[docs] Add description of sysctl parameters managed by Deckhouse (#12442)
deckhouse/deckhouse: docs/documentation/pages/security/SYSCTL.md

---
title: Sysctl parameters
permalink: en/security/sysctl.html
lang: en
---

> Deckhouse automatically configures and controls a set of the server's kernel parameters using the `sysctl` utility.

Reviewer: ```suggestion
Deckhouse automatically configures and manages a set of the server's kernel parameters using the `sysctl` utility.
```

The configured parameters improve network throughput, prevent resource depletion, and optimize memory management.

{% alert level="note" %}
If you modify these parameters, Deckhouse will automatically revert them to the values listed below.
{% endalert %}

| Parameter | Value set by Deckhouse | Description |
| -------- | ----------------------- | ----------- |
| `/sys/block/*/queue/nr_requests` | `256` | Number of queued requests for block devices. |

> | `/sys/block/*/queue/read_ahead_kb` | `256` | Amount of extra data that kernel reads from the disk to boost the reading in the future. |

Reviewer: ```suggestion
| `/sys/block/*/queue/read_ahead_kb` | `256` | Amount of extra data that kernel reads from the disk to improve future read performance. |
```

| `net.netfilter.nf_conntrack_max` | `<no-of-cores * 131072> or 524288` | Maximum number of tracked connections in the conntrack table. Calculated as "number of CPU cores" * 131072, but no lower than `524288`. |
| `net.nf_conntrack_max` | `<no-of-cores * 131072> or 524288` | Maximum number of tracked connections in the conntrack table for older kernels. Calculated as "number of CPU cores" * 131072, but no lower than `524288`. |
| `vm.dirty_background_ratio` | `5` | Percentage of system memory that can be filled with dirty pages before the kernel starts writing them to disk in the background. |
| `vm.dirty_expire_centisecs` | `12000` | Duration (in centiseconds) a dirty page can remain in system memory before it must be written to disk. |
| `vm.dirty_ratio` | `80` | Percentage of system memory that can be filled with dirty pages before all processes must stop and flush data to disk. |
| `vm.min_free_kbytes` | `131072` | Minimum amount of free memory in kilobytes reserved by the kernel for critical operations. |
| `vm.overcommit_memory` | `1` | Enables memory overcommitment. |

> | `vm.swappiness` | `0` | Disables using the swap file. |

Reviewer: ```suggestion
| `vm.swappiness` | `0` | Disables swap file usage. |
```
